Add procedure for sync meetings along with agenda template (#2)
rust-lang/leadership-council: procedures/synchronous-meetings.md

# Council Synchronous Meetings

This documents procedures, norms, and logistics of the Council's sync meetings.

## Frequency

The Council meets once a week.

After the 14th of July, 2023, the Council will re-evaluate whether this frequency still makes sense or whether it should be changed.

## Location

> The meeting happens in some video call software (e.g., Google Meet). There is a best effort to record the meeting, but this is not required since note taking is a requirement.

Reviewer: This conflicts with `## Recording`

## Quorum

The meeting has reached quorum if at *most* 2 Council representatives (and their alternates) are not present (i.e., N - 2).

If quorum is not reached, then the sync meeting is canceled and the agenda is gone through asynchronously in chat.

## Minutes

> Detailed minutes must be taken of the meeting even if the meeting was otherwise recorded. The meeting minutes must be made public within one week after the meeting concludes (or a delay must be publicly announced and resolved as soon as possible). After the meeting, Council members must review the minutes to ensure that the minutes reflect their understanding of what occurred in the meeting. Once consensus is reached that the minutes are accurate, they are released for public viewing.

Reviewer 2: Two options I've used for meetings that discuss sensitive things is redacting the meeting minutes, or simply having a vote to not release them publically.

> Detailed minutes must be taken of the meeting even if the meeting was otherwise recorded. The meeting minutes must be made public within one week after the meeting concludes (or a delay must be publicly announced and resolved as soon as possible). After the meeting, Council members must review the minutes to ensure that the minutes reflect their understanding of what occurred in the meeting. Once consensus is reached that the minutes are accurate, they are released for public viewing.

Reviewer 3: ```suggestion
Detailed minutes must be taken of the meeting even if the meeting was otherwise recorded. The meeting minutes must be made public within one week after the meeting concludes (or a delay must be publicly announced and resolved as soon as possible). After the meeting, Council members must review the minutes to ensure that the minutes reflect their understanding of what occurred in the meeting. Once there is consent that the minutes are accurate, they are released for public viewing.
```

> Detail notes must be taken of the meeting even if the meeting was otherwise recorded. The meeting notes must be made public as soon as possible (preferably latest one week after the meeting), but before doing so Council members are asked to review meeting notes to ensure that the notes reflect their understanding of what occurred in the meeting. Once consensus is reached that the meeting notes are accurate, the notes are released for public viewing. 

Reviewer: ```suggestion
Detailed notes must be taken of the meeting, even if the meeting was otherwise recorded. The meeting notes must be made public as soon as possible (preferably latest one week after the meeting), but before doing so Council members are asked to review those meeting notes to ensure that the notes reflect their understanding of what occurred in the meeting. Once consensus is reached that the meeting notes are accurate, the notes are released for public viewing. 
```

> **TODO**: decide on location for meeting notes

Reviewer 4: I have a slight preference for putting them in this repo with something like `minutes/<meeting-name>/YYYY-MM-DD.md`.

Author: I've started a poll on this [here](https://rust-lang.zulipchat.com/#narrow/stream/392734-council/topic/Home.20for.20Sync.20Meeting.20Notes/near/369595603).

## Recording

All meetings are recorded, but consensus must be reached after the meeting on what do with the recording. The Council has no obligation to do anything with the recording. If after one week, consensus on what to do with the recording has not been reached, the default is to delete the recording. 

> Recordings are usually used to ensure that the notes are correct and to optionally create automatic transcripts. Recordings can also *optionally* be made public.

Reviewer: ```suggestion
Recordings are usually used to ensure that the notes are correct and to optionally create automatic transcripts. Recordings can also *optionally* be made public, but only with consensus of all of those present on the recording.
```

> Recordings are usually used to ensure that the notes are correct and to optionally create automatic transcripts. Recordings can also *optionally* be made public, but only with consensus of all of those present on the recording.

Reviewer 5: Nit: s/consensus/consent/.

## Meeting roles

The meeting has the following assigned roles:

- **Facilitator**: the facilitator is in charge of pushing discussion along, calling on hands, formally asking the group whether there are any objections to a particular motion, and keeping the discussion on agenda.
- **Scribe(s)**: the scribes are in charge of taking notes on what was discussed in the meeting including noting any action items or follow up questions.

## Meeting Agenda

> The facilitator is in charge of ensuring that a proposal for the main agenda for the meeting is made publicly available as a pull request (as the start of the documented minutes for the meeting) to the Council team repository at least 2 days before the meeting. All Council members may comment on, revise, add to or otherwise collaborate on ensuring the agenda is in the right shape.

Reviewer 2: This seems to imply that "Facilitator" is a standing role, but then the role is assigned during the metings.
I have resolved such in my groups by having two roles "Chair" and "Speaker" - the Chair being a standing role and the *de facto* default speaker, and handling preparation o the agenda and minutes, and the Speaker being chosen during the meeting to run that meeting. Prior to a Speeker being chosen, the Chair conducts the inital items if present otherwise if they expect to be Absent, they can appoint an acting chair to conduct as much of the Chair's duties as necessary (else, the council could appoint them, perhaps synchronously).

Author: This is a good suggestion. I'm currently trying to document the status quo, but I like this suggestion and will see I can incorporate it into how we do meetings. Currently I've been playing the role of both Chair and Speaker in this scenario.

The agenda should contain the following for each agenda item:

- A short description of the contents of the item
- The champion of this agenda item
  - Usually this is the person who has suggested the agenda item and has the most context on what needs to be discussed.
- An estimated time span for the item
- A short description of the *goal(s)* of the item
- An optional proposal (something that requires review and approval through the decision making process)
  - *Note*: proposals require another Council member to second the proposal before it can be brought forward.

> - Links to previous discussion to facilitate getting up to speed with the current state of discussion around the item.

Reviewer 6: As briefly discussed today - some method of pre-raising concerns might be worth adding to agenda to identify stakeholders and topics that need discussion.

The agenda is laid out in temporal order which is determined largely by the urgency of each topic (most urgent comes first). However, this is not a hard rule as, for example, it might make sense to keep one large discussion (even if it is the most urgent) for last in the meeting so that it can simply fill the entire remainder of the meeting time.

[decisions]: http://example.com (**TODO**: link when decision making process is documented)

### Meeting structure

In addition to the main agenda items, every meeting has the following structure:

- **Check-in**
  - The check-in is a chance for Council members to connect on a human level and share how their doing outside of the context of the meeting. Examples of things shared in the check-in include: "I'm not feeling well because of my allergies, and so I'm not in a good headspace", "My partner and I just adopted a cat so I'm doing great!", and "I'm doing fine - nothing to report".
- **Assigning roles**
  - The Council members decide by consensus who will fill the roles of facilitator and note taker.
- **Turn on recording**
  - Recording is started
- **Consent to the agenda**

>   - This is a chance for last minute changes to the agenda to be surfaced and to ensure everyone is on the same page as far as what will be discussed.

Reviewer: I think this should be an opportunity to have things struck from the agenda, for example if they are not in scope or there are legal issues which might mean a discussion is unwise. I'm not sure it should be an opportunity to add items for discussion as people will not have had time to consider them in advance of the meeting. If we do allow it, the addition of a late item should only be with the consent of the council (c.f. the addition of an item earlier in the process does not require the consent of the council).

Author: Ultimately everything (items added earlier in the process as well as last minute additions and deletions) is decided through consent (as the agenda must be consented to by everyone). I've added some notes that try to encourage keeping additions small and not pushing for additions unless consent can be agreed to quickly.

- **Main agenda**
  - The main topics of the meeting which have just been consented to
- **Clarifications**
  - A call is made for any last minute clarifications, questions, requests for additional action.
- **Turn off recording**
  - The recording is stopped
- **Check-out**
  - This is a chance for council members to express how they felt the meeting went including areas for improvement, particular highlights and kudos, or any other relevant meta aspects of the meeting.

A template for this meeting structure that can be used to boostrap the agenda making process can be found [here](../templates/sync-meeting-agenda.md).

## Meeting Decorum 

The following are the social expectations that are expected of meeting participants:

- **Usage of "hands"**: all participants are expected to use the "raise hands" feature of the video meeting software to put themselves into a queue to be called on by the facilitator. There are a few exceptions to this however: the facilitator when performing the duties of that role and the scribe when asking clarifying questions.

> - **Points of order**: every participant is encouraged to raise "points of order"  which are meta commentary on some aspect of the meeting that needs clarification or discussion.

Reviewer: ```suggestion
- **Points of order**: every participant is encouraged to raise "points of order"  which are meta commentary on some aspect of the meeting that needs clarification or discussion. If a point of order is considered urgent, it may be called out verbally - bypassing the usual queuing mechanism. Such interjections should be as brief as possible, to limit talking over one another. "Point of Order - we are out of our allotted time today", might be a suitable interjection.
```

> - **Saying on topic**: while the facilitator is ultimately in charge of keeping the meeting on topic, every participant is encouraged to bring to the facilitator's attention when the conversation seems to have veered off course (either by getting off topic or by becoming too concerned with specific details).

Reviewer: ```suggestion
- **Saying on topic**: while the facilitator is ultimately in charge of keeping the meeting on topic, every participant is encouraged to bring to the facilitator's attention when the conversation seems to have veered off course (either by getting off topic or by becoming too concerned with specific details). A member may wish to raise a Point of Order to highlight this.
```

> - **Saying on topic**: while the facilitator is ultimately in charge of keeping the meeting on topic, every participant is encouraged to bring to the facilitator's attention when the conversation seems to have veered off course (either by getting off topic or by becoming too concerned with specific details).

Reviewer 4: ?

```suggestion
- **Staying on topic**: while the facilitator is ultimately in charge of keeping the meeting on topic, every participant is encouraged to bring to the facilitator's attention when the conversation seems to have veered off course (either by getting off topic or by becoming too concerned with specific details).
```

# Council Synchronous Meetings

This documents procedures, norms, and logistics of the Council's sync meetings.

## Frequency

The Council meets once a week.

> After the 14th of July, 2023, the Council will re-evaluate whether this frequency still makes sense or whether it should be changed.

Reviewer 3: I'd prefer to have an upper bound on this date as well. Could we say "on the first council meeting on or after July 14, 2023, we will re-evaluate this frequency?"

## Location

> The meeting happens in some video call software (e.g., Google Meet).

Reviewer 3: Once this is merged, maybe we should raise an issue to get a little more concrete here. I'm not picky about the software, but it'd be good to have a default.

## Quorum

> The meeting has reached quorum if at *most* 2 Council representatives (and their alternates) are not present (i.e., N - 2).

Reviewer 3: This is probably a different policy question, but have we talked about how we pick alternates? Are alternates a standing position, or is this something the council member can appoint as needed?

Reviewer 4: I believe the RFC indicates that the top-level team is responsible for choosing an alternate ([source](https://rust-lang.github.io/rfcs/3392-leadership-council.html#alternates-and-forgoing-representation)).

So, I'd say the selection process is up to the team. And whether they select ahead-of-time, or on an as-needed basis is also presumably up to the team.

Presumably if a council member knows they are going to be unavailable, they are responsible for letting their team know and get the team to pick someone. If the council member is becomes unavailable without communicating, the council should inform the team.

I'm personally unsure if there should be standing alternates or not. It seems like it might be a good idea just so there isn't a scramble to pick someone. But I could imagine people may not want to be on the hook with uncertainty if they will actually ever be needed.

Author: Yes, this was something the governance reform working group explicitly did not want to dictate or have the Council dictate to the top-level teams since it is their responsibility to select representatives how they see fit. However, we *could* offer "best practices" for teams to adopt which they may deviate from if they wish to. Thoughts?

Reviewer 5: If it's meant to be a softer recommendation, I'd prefer to not have it firmly documented, as what we put here will carry weight that may be incorrectly used.

Reviewer 3: I agree with the idea of leaving things up to the team where possible. Having some suggestions, like "we recommend teams do X but teams are free to make their own choices based on their needs" would be okay with me, but I think it makes sense to hold off on making these recommendations for now.

Author: Going to resolve this as it's not really relevant to this document. Happy to see this logged somewhere else though.

If quorum is not reached, then the sync meeting is canceled and the agenda is gone through asynchronously in chat.

## Minutes

> Detailed minutes must be taken of the meeting even if the meeting was otherwise recorded. The meeting notes must be made public as soon as possible (preferably latest one week after the meeting), but before doing so Council members are asked to review those meeting notes to ensure that the notes reflect their understanding of what occurred in the meeting. Once consensus is reached that the meeting notes are accurate, the notes are released for public viewing. 

Reviewer 3: ```suggestion
Detailed minutes must be taken of the meeting even if the meeting was otherwise recorded. The meeting notes must be made public as soon as possible (preferably at most one week after the meeting), but before doing so Council members are asked to review those meeting notes to ensure that the notes reflect their understanding of what occurred in the meeting. Once consensus is reached that the meeting notes are accurate, the notes are released for public viewing. 
```

> Detailed minutes must be taken of the meeting even if the meeting was otherwise recorded. The meeting notes must be made public as soon as possible (preferably latest one week after the meeting), but before doing so Council members are asked to review those meeting notes to ensure that the notes reflect their understanding of what occurred in the meeting. Once consensus is reached that the meeting notes are accurate, the notes are released for public viewing. 

Reviewer 3: I'd like to make "as soon as possible" a little more concrete, more to the level of "we promise to release minutes within X time and if not then we have not achieved one of our commitments to the community." I know this may not always be possible, but it'd be nice to be able to produce statistics about how often we meet this commitment. I see making concrete time commitments as a way of building trust with the community because over time we can point to a consistent track record of committing to things and following through.

Author: I tried to make this more concrete by at least requiring a public announcement when the minutes are delayed. I think that's a good first step. We'll see how this procedure works out in practice, and we can adjust from there.

**TODO**: decide on location for meeting notes

## Recording

All meetings are recorded, but consensus must be reached after the meeting on what do with the recording. The Council has no obligation to do anything with the recording. If after one week, consensus on what to do with the recording has not been reached, the default is to delete the recording. 

Recordings are usually used to ensure that the notes are correct and to optionally create automatic transcripts. Recordings can also *optionally* be made public, but only with consensus of all of those present on the recording.

## Meeting roles

The meeting has the following assigned roles:

- **Facilitator**: the facilitator is in charge of pushing discussion along, calling on hands, formally asking the group whether there are any objections to a particular motion, and keeping the discussion on agenda.

> - **Scribe(s)**: the scribes are in charge of taking notes on what was discussed in the meeting including noting any action items or follow up questions.

Reviewer 3: This may be better as a follow-up policy document, but I liked @Manishearth's description of the scribing process they've used in W3C meetings (I've lost the Zulip link for where this was discussed). The idea was to have one person in charge of scribing, other's contribute, and when the scribe is called on to speak there's an explicit delegation to an alternate scribe while the main scribe speaks.

Author: This sounds great. I'd like for us to try this out and adopt the process before documenting it as the process though. I'll put this in the agenda for the next meeting to mention that we might want to try something like this.

## Meeting Agenda

> The facilitator is in charge of ensuring that a proposal for the main agenda for the meeting is written into the notes document at least 2 days before the meeting. All Council members may comment on, revise, add to or otherwise collaborate on ensuring the agenda is in the right shape.

Reviewer 3: Since currently our meetings haven't been open to the public, I think it'd be good if the agenda document could be public before the meeting so people can know what we're going to be talking about, and also to give council members a chance to get input from their team before the meeting.

Author: Good point - the agenda's have been public already, but we should document them as such. We should probably figure out standard place to post these so folks can more easily find them. I've stated that the agenda shall be posted as a PR to this repo (which I think makes sense we should have the agenda above meeting minutes).

The agenda should contain the following for each agenda item:

- A short description of the contents of the item
- The champion of this agenda item
  - Usually this is the person who has suggested the agenda item and has the most context on what needs to be discussed.
- An estimated time span for the item

> - A short description of the *goal(s)* of the item

Reviewer 3: In meetings I've had with @yaahc we've prefixed items with a tag like (I), (E), (D), which means inform, explore, or decide. I've found this helpful to make it clear at a glance what the purpose of the discussion is.

> - A short description of the *goal(s)* of the item

Reviewer 7: Oooooh I like that.

> - A short description of the *goal(s)* of the item

Author: I like this as well. We can give this a try and see how it sticks though I might avoid usage of abbreviations to make sure the agendas are understandable by those unfamiliar with the procedure.

- An optional proposal (something that requires review and approval through the [decision making process][decisions])
  - *Note*: proposals require another Council member to second the proposal before it can be brought forward.
- Links to previous discussion to facilitate getting up to speed with the current state of discussion around the item.

The agenda is laid out in temporal order which is determined largely by the urgency of each topic (most urgent comes first). However, this is not a hard rule as, for example, it might make sense to keep one large discussion (even if it is the most urgent) for last in the meeting so that it can simply fill the entire remainder of the meeting time.

> [decisions]: http://example.com (**TODO**: link when decision making process is documented)

Author: I've removed this. We'll need to make a habit of going back to older documents and make sure things are well linked so I don't think this needs an explicit TODO.

- **Main agenda**
  - The main topics of the meeting which have just been consented to
- **Clarifications**
  - A call is made for any last minute clarifications, questions, requests for additional action.
- **Turn off recording**
  - The recording is stopped
- **Check-out**
  - This is a chance for council members to express how they felt the meeting went including areas for improvement, particular highlights and kudos, or any other relevant meta aspects of the meeting.

A template for this meeting structure that can be used to boostrap the agenda making process can be found [here](../templates/sync-meeting-agenda.md).

## Meeting Decorum 

The following are the social expectations that are expected of meeting participants:

- **Usage of "hands"**: all participants are expected to use the "raise hands" feature of the video meeting software to put themselves into a queue to be called on by the facilitator. There are a few exceptions to this however: the facilitator when performing the duties of that role and the scribe when asking clarifying questions.

> - **Points of order**: every participant is encouraged to raise "points of order"  which are meta commentary on some aspect of the meeting that needs clarification or discussion. If a point of order is considered urgent, it may be called out verbally - bypassing the usual queuing mechanism. Such interjections should be as brief as possible, to limit talking over one another. For example, "Point of Order - we are out of our allotted time today", might be a suitable interjection.

Reviewer 3: This says that for urgent points of order they can be made verbally. Is the implication that non-urgent points of order (are their any?) should be raised through the chat instead?

More generally, do we want to add an expectation around use of chat during the meeting? In a lot of calls I'm on we use the chat basically to have side discussions, add clarifying points, or short comments that don't warrant raising a hand. So far we haven't done much of that in council meetings, and I kind of like that. I think it's easier for us all to follow the discussion if we don't have a side stream discussion going.

Author: Yea I think having side conversations in chat is not a good idea, but yes, I think we have been using the chat for non-urgent points of order and I think we should continue doing so.
